lib/ansible/playbook/conditional.py

```diff
@@ -182,12 +182,8 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
                             inside_yield=inside_yield
                         )
             try:
-                e = templar.environment.overlay()
-                e.filters.update(templar.environment.filters)
-                e.tests.update(templar.environment.tests)
-
-                res = e._parse(conditional, None, None)
-                res = generate(res, e, None, None)
+                res = templar.environment._parse(conditional, None, None)
```

Comment:
You didn't change this, but should we use something other than a private method (`._parse`) here?  Not sure why we haven't just used `Environment.parse` from the beginning, other than it does some exception handling.

Reply:
Good point. I added a FIXME there.

Reply:
Sounds good. Done.